flexmeasures/cli/data_edit.py

```diff
@@ -0,0 +1,241 @@
+)
+@click.option(
+    "--sensor-id",
+    "sensors",
+    required=False,
+    multiple=True,
+    type=SensorField(),
+    help="Add/edit attribute to this sensor. Follow up with the sensor's ID.",
+)
+@click.option(
+    "--attribute",
+    "attribute_key",
+    required=True,
+    help="Add/edit this attribute. Follow up with the name of the attribute.",
+)
+@click.option(
+    "--float",
+    "attribute_float_value",
+    required=False,
+    type=float,
+    help="Set the attribute to this float value.",
+)
+@click.option(
+    "--bool",
+    "attribute_bool_value",
+    required=False,
+    type=bool,
+    help="Set the attribute to this bool value.",
+)
+@click.option(
+    "--str",
+    "attribute_str_value",
+    required=False,
+    type=str,
+    help="Set the attribute to this string value.",
+)
+@click.option(
+    "--int",
+    "attribute_int_value",
+    required=False,
+    type=int,
+    help="Set the attribute to this integer value.",
+)
+@click.option(
+    "--null",
+    "attribute_null_value",
+    required=False,
+    is_flag=True,
+    default=False,
+    help="Set the attribute to a null value.",
+)
+def edit_asset_attribute(
+    attribute_key: str,
+    assets: List[GenericAsset],
+    sensors: List[Sensor],
+    attribute_null_value: bool,
+    attribute_float_value: Optional[float] = None,
+    attribute_bool_value: Optional[bool] = None,
+    attribute_str_value: Optional[str] = None,
+    attribute_int_value: Optional[int] = None,
+):
+    """Edit (or add) an asset attribute or sensor attribute."""
+
+    if not assets and not sensors:
+        raise ValueError("Missing flag: pass at least one --asset-id or --sensor-id.")
+
+    # Parse attribute value
+    attribute_value = parse_attribute_value(
+        attribute_float_value=attribute_float_value,
+        attribute_bool_value=attribute_bool_value,
+        attribute_str_value=attribute_str_value,
+        attribute_int_value=attribute_int_value,
+        attribute_null_value=attribute_null_value,
+    )
+
+    # Set attribute
+    for asset in assets:
+        asset.attributes[attribute_key] = attribute_value
+        db.session.add(asset)
+    for sensor in sensors:
+        sensor.attributes[attribute_key] = attribute_value
+        db.session.add(sensor)
+    db.session.commit()
+
+
+@fm_edit_data.command("resample-data")
+@with_appcontext
+@click.option(
+    "--sensor-id",
+    "sensor_ids",
+    multiple=True,
+    required=True,
+    help="Resample data for this sensor. Follow up with the sensor's ID. This argument can be given multiple times.",
+)
+@click.option(
+    "--event-resolution",
+    "event_resolution_in_minutes",
+    type=int,
+    required=True,
+    help="New event resolution as an integer number of minutes.",
+)
+@click.option(
+    "--from",
+    "start_str",
+    required=False,
+    help="Resample only data from this datetime onwards. Follow up with a timezone-aware datetime in ISO 6801 format.",
+)
+@click.option(
+    "--until",
```

Comment:
This probably would also be less work if we used `--duration`.

Reply:
Let me know if I should also do that, as I did in #371 :)

Reply:
In this particular case, I don't think it makes sense to specify a duration. I believe it would be more prone to produce mistakes. If I know that some sensor switched its resolution at some point in time and want to resample the preceding data, I want to specify that point in time directly, instead of having to look up the start of the data set, specify that, calculate the duration between the time of switching and the start, and specify that.